benchmark/tcp_bench_test.go

```diff
@@ -46,17 +47,26 @@ func echoServer(tb testing.TB) net.Listener {
 
 func benchmarkClient(b *testing.B, dst string, reqSize int) {
 	req := getRequestBytes(reqSize)
+	totalExpected := b.N * reqSize
 
 	conn, err := net.Dial("tcp", dst)
 	require.NoError(b, err, "Failed to connect to destination")
+	defer conn.Close()
 
-	go io.Copy(ioutil.Discard, conn)
+	readerDone := make(chan struct{})
+	go func() {
+		defer close(readerDone)
+		_, err := io.CopyN(ioutil.Discard, conn, int64(totalExpected))
+		assert.NoError(b, err, "Expected more response bytes")
```

Comment:
The error message might be nicer if we captured the number of bytes successfully written ("Expected 1024 response bytes, got 0").